src/test/java/ArrayQueueTest.java

```diff
@@ -1,101 +1,208 @@
+                }
+
+
+                @Test
+                void enqueue_successfullyAdds_returnsTrue() {
+                    assertTrue(classUnderTest.enqueue(11));
+                }
+
+                @Test
+                void enqueue_many_unchangedFirst() {
+                    classUnderTest.enqueue(11);
+                    assertEquals(10, classUnderTest.first());
+                }
+
+                @Test
+                void dequeue_many_returnsFirst() {
+                    assertEquals(10, classUnderTest.dequeue());
+                }
+
+                @Test
+                void dequeue_many_newFirst() {
+                    classUnderTest.dequeue();
+                    assertEquals(20, classUnderTest.first());
+                }
+
+                @Test
+                void first_many_returnsFirst() {
+                    assertEquals(10, classUnderTest.first());
+                }
+
+                @Test
+                void first_many_unchanged() {
+                    classUnderTest.first();
+                    assertEquals(preState, classUnderTest);
+                }
+
+                @Test
+                void isEmpty_many_returnsFalse() {
+                    assertFalse(classUnderTest.isEmpty());
+                }
+
+                @Test
+                void size_many_returnsCorrectSize() {
+                    assertEquals(4, classUnderTest.size());
+                }
+
+                @Test
+                void toString_singleton_returnsCorrectString() {
+                    assertEquals("10, 20, 30, 40, ", classUnderTest.toString());
+                }
+            }
         }
     }
 
-    @Test
-    @DisplayName("Dequeuing throws NoSuchElementException when queue is empty.")
-    void dequeueEmptyQueueThrowsException() {
-        Queue<Integer> queue = new ArrayQueue<>();
-        assertThrows(NoSuchElementException.class, () -> queue.dequeue());
-    }
+    @Nested
+    class WhenLarge {
+        @Test
+        void enqueue_large_successfullyExpandsCapacity() {
+            for (int i = 0; i < 1000; i++) {
+                classUnderTest.enqueue(i);
+            }
+            assertEquals(1000, classUnderTest.size());
+        }
 
-    @Test
-    @DisplayName("First throws NoSuchElementException when queue is empty.")
-    void firstEmptyQueueThrowsException() {
-        Queue<Integer> queue = new ArrayQueue<>();
-        assertThrows(NoSuchElementException.class, () -> queue.first());
+        @Test
+        void dequeue_large_fifoOrdering() {
+            for (int i = 0; i < 1000; i++) {
+                classUnderTest.enqueue(i);
+            }
+            for (int i = 0; i < 500; i++) {
+                classUnderTest.dequeue();
+            }
+            for (int i = 1000; i < 1500; i++) {
+                classUnderTest.enqueue(i);
+            }
+            for (int i = 500; i < 1500; i++) {
+                assertEquals(i, classUnderTest.dequeue());
+            }
+        }
```

Comment:
Definitely remove. This is not providing any new information.

Reply:
This would then apply to the suite of other implementations sitting in PRs right now.